### ISSUE TITLE
SNMP trap/inform receiver

### ISSUE THREAD

**Issue by abarik1981** [markabrahams/node-net-snmp#3]
When do you think will node net-snmp will support SNMP trap listening? Great job by the way

**Comment by stephenwvickers** [comment]
Hi,

This is something I would love to have done already, but just haven't been given the time yet.

Unfortunately I cannot give an estimate on time as I am very busy with work.

I will leave this issue open and update it if I have any further information.

Thanks

Steve

**Comment by kamakazikamikaze** [comment]
Just wanted to also say that this is a feature I'd love to see. Great work so far!

**Comment by out4b** [comment]
Thanks for the great work. Would love to see this feature enabled.

**Comment by markabrahams** [comment]
@stephenwvickers - I've implemented the first cut of a trap/inform receiver that handles v1 traps, v2 traps, and receives and responds to v2 informs.  I haven't done SNMPv3 trap handling yet, but will probably finish that in the next day or two.  I won't put a PR in for this until we've discussed the SNMPv3 PR (#53) as the trap/inform receiver depends on PR #53.  I'm free for the next few hours if you're available, or if that doesn't suit, feel free to touch base at any stage.  Cheers!

**Comment by markabrahams** [comment]
@stephenwvickers - FYI - I've finished the trap/inform receiver now.  It has complete support for all trap and inform notification scenarios: v1 trap PDU support for SNMPv1, and v2 trap PDU and inform request PDU support for v2c and v3, with noAuthNoPriv, authNoPriv, and authPriv supported for v3 traps and informs.  This is currently sitting in a "receiver" branch on my fork.  As mentioned, I'll hold off raising a PR for this until we sort out the SNMPv3 PR #53, as this is obviously a dependency for the v3 components of the receiver.

**Comment by markabrahams** [comment]
I've also merged this into the master branch on my fork.  So this can be tested pre-merge (pre-PR in fact!) by pointing your package.json:
"net-snmp": "https://github.com/markabrahams/node-net-snmp.git"

and then "npm install"

There's a new Receiver class for this feature, with the main public API call:
snmp.createReceiver(options, callback)

If you use the disableAuthorization option (and stick to v1, v2c or v3 noAuthNoPriv), it'll then start hitting your callback as traps/informs hit the Receiver, with no further configuration required.  If you don't disable authorization (which is the default), there are a few more calls to manage the authorization lists.

Full documentation for using the Receiver is at:
https://github.com/markabrahams/node-net-snmp

**Comment by markabrahams** [comment]
This code is merged into master now, and also published as version 2.1.0 of the net-snmp NPM.
